src/LAND/LAND.router.ts

```diff
@@ -0,0 +1,204 @@
+import { env } from 'decentraland-commons'
+import { server } from 'decentraland-server'
+import FormData from 'form-data'
+import { Request } from 'express'
+//@ts-ignore
+import * as contentHash from 'content-hash'
+import fetch, { Response as FetchResponse } from 'node-fetch'
+import { HTTPError, STATUS_CODES } from '../common/HTTPError'
+import { Router } from '../common/Router'
+import { getCID } from '../utils/cid'
+import {
+  GetRedirectionHashesResponse,
+  UploadRedirectionResponse,
+} from './LAND.types'
+
+export const MAX_COORDS = 150
+const INDEX_FILE = 'index.html'
+
+export class LANDRouter extends Router {
+  private ipfsUrl = env.get('IPFS_URL')
+  private ipfsProjectId = env.get('IPFS_PROJECT_ID')
+  private ipfsApiKey = env.get('IPFS_API_KEY')
+  private explorerUrl = env.get('EXPLORER_URL')
+
+  mount() {
+    this.verifyEnvs()
+
+    this.router.get(
+      '/lands/redirectionHashes',
+      server.handleRequest(this.getRedirectionHashes)
+    )
+
+    this.router.post(
+      '/lands/:coords/redirection',
+      server.handleRequest(this.uploadRedirection)
+    )
+  }
+
+  private uploadRedirection = async (
+    req: Request
+  ): Promise<UploadRedirectionResponse> => {
+    const coords = server.extractFromReq(req, 'coords')
+
+    this.validateCoords(coords)
+
+    const locale = req.headers['accept-language']
+
+    const redirectionFile = this.generateRedirectionFile(coords, locale)
+
+    const formData = new FormData()
+
+    formData.append('blob', redirectionFile, INDEX_FILE)
+
+    let result: FetchResponse
+
+    try {
+      result = await fetch(this.ipfsUrl + '/api/v0/add?pin=false', {
+        method: 'post',
+        body: formData,
+        headers: {
+          Authorization: `Basic ${Buffer.from(
+            `${this.ipfsProjectId}:${this.ipfsApiKey}`
+          ).toString('base64')}`,
+        },
+      })
+    } catch (e) {
+      throw new HTTPError(
+        'Failed to upload file to IPFS # Request',
+        {},
+        STATUS_CODES.error
+      )
+    }
+
+    if (!result.ok) {
+      throw new HTTPError(
+        'Failed to upload file to IPFS # Response',
+        {},
+        STATUS_CODES.error
+      )
+    }
+
+    const { Hash } = await result.json()
+
+    return {
+      ipfsHash: Hash,
+      contentHash: await contentHash.fromIpfs(Hash),
+    }
+  }
+
+  private getRedirectionHashes = async (
+    req: Request
+  ): Promise<GetRedirectionHashesResponse> => {
+    let coordsListQP: string | string[]
+
+    try {
+      coordsListQP = server.extractFromReq<string | string[]>(req, 'coords')
+    } catch (e) {
+      throw new HTTPError(e.message, {}, STATUS_CODES.badRequest)
+    }
+
+    const coordsList =
+      typeof coordsListQP === 'string' ? [coordsListQP] : coordsListQP
+
+    if (coordsList.length > MAX_COORDS) {
+      throw new HTTPError(
+        `Max ${MAX_COORDS} coords`,
+        { amount: coordsList.length },
+        STATUS_CODES.badRequest
+      )
+    }
```

Comment:
I'm not sure if this is necessary taking into consideration the 2048 hard limit of the URL's length. We can still have the max length, but it might be redundant.

Reply:
We're keeping the limit.